lighthouse-core/config/config.js

```diff
@@ -330,6 +319,121 @@ class Config {
+    config.passes = Config.getPassesNeededByGatherers(config.passes, requiredGatherers);
+    return config;
+  }
+
+ /**
+  * Return names of top-level aggregations from a config. Used by tools driving lighthouse-core
+  * @param {!Object} config Lighthouse config object.
+  * @return {!Array<string>}  Name values of aggregations within
+  */
+  static getAggregationNames(config) {
+    return config.aggregations.map(agg => agg.name);
+  }
+
+  /**
+   * Find audits required for remaining aggregations.
+   * @param  {!Object} aggregations
+   * @return {!Set<string>}
+   */
+  static getAuditsNeededByAggregations(aggregations) {
+    const requestedItems = _flatten(aggregations.map(aggregation => aggregation.items));
+    const requestedAudits = _flatten(requestedItems.map(item => Object.keys(item.audits)));
+    return new Set(requestedAudits);
+  }
+
+  /**
+   * Creates mapping from audit path (used in config.audits) to audit.name (used in config.aggregations)
+   * @param {!Object} config Lighthouse config object.
+   * @return {Map}
+   */
+  static getMapOfAuditPathToName(config) {
+    // The `audits` property in the config is a list of paths of audits to run.
+    // `requestedAuditNames` is a list of audit *names*. Map paths to names, then
+    // filter out any paths of audits with names that weren't requested.
+    const auditObjectsAll = Config.requireAudits(config.audits);
+    const auditPathToName = new Map(auditObjectsAll.map((AuditClass, index) => {
+      const auditPath = config.audits[index];
+      const auditName = AuditClass.meta.name;
+      return [auditPath, auditName];
+    }));
+    return auditPathToName;
+  }
+
+  /**
+   * From some requested audits, return names of all required artifacts
+   * @param  {!Object} audits
+   * @return {!Set<string>}
+   */
+  static getGatherersNeededByAudits(audits) {
+    // It's possible we weren't given any audits (but existing audit results), in which case
+    // there is no need to do any work here.
+    if (!audits) {
+      return new Set();
+    }
+
+    return audits.reduce((list, audit) => {
+      audit.meta.requiredArtifacts.forEach(artifact => list.add(artifact));
+      return list;
+    }, new Set());
+  }
+
+  /**
+   * Filters to only required passes and gatherers, returning a new passes object
+   * @param  {!Object} passes
+   * @param  {!Set<string>} requiredGatherers
+   * @return {!Object} fresh passes object
+   */
+  static getPassesNeededByGatherers(passes, requiredGatherers) {
+    const passesClone = JSON.parse(JSON.stringify(passes));
+    const filteredPasses = passesClone.map(pass => {
+      // remove any unncessary gatherers
+      pass.gatherers = pass.gatherers.filter(gathererName => {
+        gathererName = GatherRunner.getGathererClass(gathererName).name;
+        return requiredGatherers.has(gathererName);
+      });
+      return pass;
+    // remove any passes lacking concrete gatherers
+    }).filter(pass => pass.gatherers.length > 0);
```

Comment:
we can eliminate recordNetwork ones but not recordTrace ones. this will require some better handling, but i'm not addressing it now.

logic updated to keep recordTrace ones, otherwise nuke pass if no gatherers.